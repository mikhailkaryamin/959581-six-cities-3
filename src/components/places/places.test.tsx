import React from "react";
import renderer from "react-test-renderer";
import configureStore from "redux-mock-store";
import {
  Provider
} from "react-redux";
import {
  BrowserRouter
} from 'react-router-dom';
import {
  storeWithAuth
} from "../../mocks/initialState";
import {
  ClassModificator
} from "../../consts";
import Places from "./places";

const mockStore = configureStore([]);

const currentCityOffers = [
  {
    bedrooms: 3,
    city: {
      location: {
        latitude: 52.370216,
        longitude: 4.895168,
        zoom: 10
      },
      name: `Amsterdam`
    },
    description: `A quiet cozy and picturesque that hides behind a a river by the unique lightness of Amsterdam.`,
    goods: [`Heating`, `Kitchen`, `Cable TV`, `Washing machine`, `Coffee machine`, `Dishwasher`],
    host: {
      avatarUrl: `img/1.png`,
      id: 3,
      isPro: true,
      name: `Angelina`
    },
    id: 1,
    images: [`img/1.png`, `img/2.png`],
    isFavorite: false,
    isPremium: false,
    location: {
      latitude: 52.35514938496378,
      longitude: 4.673877537499948,
      zoom: 8
    },
    maxAdults: 4,
    previewImage: `img/1.png`,
    price: 120,
    rating: 4.8,
    title: `Beautiful & luxurious studio at great location`,
    type: `apartment`
  },
  {
    bedrooms: 3,
    city: {
      location: {
        latitude: 52.370216,
        longitude: 4.895168,
        zoom: 10
      },
      name: `Amsterdam`
    },
    description: `A quiet cozy and picturesque that hides behind a a river by the unique lightness of Amsterdam.`,
    goods: [`Heating`, `Kitchen`, `Cable TV`, `Washing machine`, `Coffee machine`, `Dishwasher`],
    host: {
      avatarUrl: `img/1.png`,
      id: 3,
      isPro: true,
      name: `Angelina`
    },
    id: 2,
    images: [`img/1.png`, `img/2.png`],
    isFavorite: false,
    isPremium: false,
    location: {
      latitude: 52.35514938496378,
      longitude: 4.673877537499948,
      zoom: 8
    },
    maxAdults: 4,
    previewImage: `img/1.png`,
    price: 120,
    rating: 4.8,
    title: `Beautiful & luxurious studio at great location`,
    type: `apartment`
  },
  {
    bedrooms: 3,
    city: {
      location: {
        latitude: 52.370216,
        longitude: 4.895168,
        zoom: 10
      },
      name: `Amsterdam`
    },
    description: `A quiet cozy and picturesque that hides behind a a river by the unique lightness of Amsterdam.`,
    goods: [`Heating`, `Kitchen`, `Cable TV`, `Washing machine`, `Coffee machine`, `Dishwasher`],
    host: {
      avatarUrl: `img/1.png`,
      id: 3,
      isPro: true,
      name: `Angelina`
    },
    id: 3,
    images: [`img/1.png`, `img/2.png`],
    isFavorite: false,
    isPremium: false,
    location: {
      latitude: 52.35514938496378,
      longitude: 4.673877537499948,
      zoom: 8
    },
    maxAdults: 4,
    previewImage: `img/1.png`,
    price: 120,
    rating: 4.8,
    title: `Beautiful & luxurious studio at great location`,
    type: `apartment`
  },
  {
    bedrooms: 3,
    city: {
      location: {
        latitude: 52.370216,
        longitude: 4.895168,
        zoom: 10
      },
      name: `Amsterdam`
    },
    description: `A quiet cozy and picturesque that hides behind a a river by the unique lightness of Amsterdam.`,
    goods: [`Heating`, `Kitchen`, `Cable TV`, `Washing machine`, `Coffee machine`, `Dishwasher`],
    host: {
      avatarUrl: `img/1.png`,
      id: 3,
      isPro: true,
      name: `Angelina`
    },
    id: 4,
    images: [`img/1.png`, `img/2.png`],
    isFavorite: false,
    isPremium: false,
    location: {
      latitude: 52.35514938496378,
      longitude: 4.673877537499948,
      zoom: 8
    },
    maxAdults: 4,
    previewImage: `img/1.png`,
    price: 120,
    rating: 4.8,
    title: `Beautiful & luxurious studio at great location`,
    type: `apartment`
  }
];
const currentSort = `Popular`;
const currentCity = `Paris`;
const onCardHover = () => {};
const onCardLeave = () => {};
const handleSortChange = () => {};

it(`Render places (modificator - cities) correctly`, () => {
  const store = mockStore(storeWithAuth);
  const tree = renderer
    .create(
        <Provider
          store={store}
        >
          <BrowserRouter>
            <Places
              classModificator={ClassModificator.CITIES_PLACES}
              currentCityOffers={currentCityOffers}
              currentCity={currentCity}
              currentSort={currentSort}
              onCardHover={onCardHover}
              handleSortChange={handleSortChange}
              onCardLeave={onCardLeave}
            />
          </BrowserRouter>
        </Provider>
    ).toJSON();

  expect(tree).toMatchSnapshot();
});

it(`Render places (modificator - near) correctly`, () => {
  const store = mockStore(storeWithAuth);
  const tree = renderer
    .create(
        <Provider
          store={store}
        >
          <BrowserRouter>
            <Places
              classModificator={ClassModificator.NEAR_PLACES}
              currentCityOffers={currentCityOffers}
              currentSort={currentSort}
              onCardHover={onCardHover}
              onCardLeave={onCardLeave}
            />
          </BrowserRouter>
        </Provider>

    ).toJSON();

  expect(tree).toMatchSnapshot();
});
