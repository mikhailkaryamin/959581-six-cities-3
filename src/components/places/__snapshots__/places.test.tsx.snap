// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Render places (modificator - cities) correctly 1`] = `
<section
  className="places cities__places"
>
  <h2
    className="visually-hidden"
  >
    Places
  </h2>
  <b
    className="places__found"
  >
    4
     places to stay in 
    Paris
  </b>
  <form
    action="#"
    className="places__sorting"
    method="get"
    onClick={[Function]}
  >
    <span
      className="places__sorting-caption"
    >
      Sort by 
    </span>
    <span
      className="places__sorting-type"
      tabIndex={0}
    >
      Popular
      <svg
        className="places__sorting-arrow"
        height="4"
        width="7"
      >
        <use
          xlinkHref="#icon-arrow-select"
        />
      </svg>
    </span>
    <ul
      className="places__options places__options--custom "
    >
      <li
        className="places__option"
        onClick={[Function]}
        tabIndex={0}
      >
        Price: low to high
      </li>
      <li
        className="places__option"
        onClick={[Function]}
        tabIndex={0}
      >
        Price: high to low
      </li>
      <li
        className="places__option"
        onClick={[Function]}
        tabIndex={0}
      >
        Top rated first
      </li>
    </ul>
  </form>
  <div
    className="places__list cities__places-list tabs__content"
  >
    <article
      className="cities__place-card place-card"
      onMouseEnter={[Function]}
      onMouseLeave={[Function]}
    >
      <div
        className="cities__image-wrapper place-card__image-wrapper"
      >
        <a
          href="#"
        >
          <img
            alt="Place image"
            className="place-card__image"
            height={200}
            src="img/1.png"
            width={260}
          />
        </a>
      </div>
      <div
        className="place-card__info"
      >
        <div
          className="place-card__price-wrapper"
        >
          <div
            className="place-card__price"
          >
            <b
              className="place-card__price-value"
            >
              €
              120
            </b>
            <span
              className="place-card__price-text"
            >
              / night
            </span>
          </div>
          <button
            className="cities__bookmark-button button "
            onClick={[Function]}
            type="button"
          >
            <svg
              className="place-card__bookmark-icon"
              height={19}
              width={18}
            >
              <use
                xlinkHref="#icon-bookmark"
              />
            </svg>
            <span
              className="visually-hidden"
            >
              To bookmarks
            </span>
          </button>
        </div>
        <div
          className="place-card__rating rating"
        >
          <div
            className="place-card__stars rating__stars"
          >
            <span
              style={
                Object {
                  "width": "100%",
                }
              }
            />
            <span
              className="visually-hidden"
            >
              Rating
            </span>
          </div>
        </div>
        <h2
          className="place-card__name"
        >
          <a
            href="/offer/1"
            onClick={[Function]}
          >
            Beautiful & luxurious studio at great location
          </a>
        </h2>
        <p
          className="place-card__type"
        >
          apartment
        </p>
      </div>
    </article>
    <article
      className="cities__place-card place-card"
      onMouseEnter={[Function]}
      onMouseLeave={[Function]}
    >
      <div
        className="cities__image-wrapper place-card__image-wrapper"
      >
        <a
          href="#"
        >
          <img
            alt="Place image"
            className="place-card__image"
            height={200}
            src="img/1.png"
            width={260}
          />
        </a>
      </div>
      <div
        className="place-card__info"
      >
        <div
          className="place-card__price-wrapper"
        >
          <div
            className="place-card__price"
          >
            <b
              className="place-card__price-value"
            >
              €
              120
            </b>
            <span
              className="place-card__price-text"
            >
              / night
            </span>
          </div>
          <button
            className="cities__bookmark-button button "
            onClick={[Function]}
            type="button"
          >
            <svg
              className="place-card__bookmark-icon"
              height={19}
              width={18}
            >
              <use
                xlinkHref="#icon-bookmark"
              />
            </svg>
            <span
              className="visually-hidden"
            >
              To bookmarks
            </span>
          </button>
        </div>
        <div
          className="place-card__rating rating"
        >
          <div
            className="place-card__stars rating__stars"
          >
            <span
              style={
                Object {
                  "width": "100%",
                }
              }
            />
            <span
              className="visually-hidden"
            >
              Rating
            </span>
          </div>
        </div>
        <h2
          className="place-card__name"
        >
          <a
            href="/offer/2"
            onClick={[Function]}
          >
            Beautiful & luxurious studio at great location
          </a>
        </h2>
        <p
          className="place-card__type"
        >
          apartment
        </p>
      </div>
    </article>
    <article
      className="cities__place-card place-card"
      onMouseEnter={[Function]}
      onMouseLeave={[Function]}
    >
      <div
        className="cities__image-wrapper place-card__image-wrapper"
      >
        <a
          href="#"
        >
          <img
            alt="Place image"
            className="place-card__image"
            height={200}
            src="img/1.png"
            width={260}
          />
        </a>
      </div>
      <div
        className="place-card__info"
      >
        <div
          className="place-card__price-wrapper"
        >
          <div
            className="place-card__price"
          >
            <b
              className="place-card__price-value"
            >
              €
              120
            </b>
            <span
              className="place-card__price-text"
            >
              / night
            </span>
          </div>
          <button
            className="cities__bookmark-button button "
            onClick={[Function]}
            type="button"
          >
            <svg
              className="place-card__bookmark-icon"
              height={19}
              width={18}
            >
              <use
                xlinkHref="#icon-bookmark"
              />
            </svg>
            <span
              className="visually-hidden"
            >
              To bookmarks
            </span>
          </button>
        </div>
        <div
          className="place-card__rating rating"
        >
          <div
            className="place-card__stars rating__stars"
          >
            <span
              style={
                Object {
                  "width": "100%",
                }
              }
            />
            <span
              className="visually-hidden"
            >
              Rating
            </span>
          </div>
        </div>
        <h2
          className="place-card__name"
        >
          <a
            href="/offer/3"
            onClick={[Function]}
          >
            Beautiful & luxurious studio at great location
          </a>
        </h2>
        <p
          className="place-card__type"
        >
          apartment
        </p>
      </div>
    </article>
    <article
      className="cities__place-card place-card"
      onMouseEnter={[Function]}
      onMouseLeave={[Function]}
    >
      <div
        className="cities__image-wrapper place-card__image-wrapper"
      >
        <a
          href="#"
        >
          <img
            alt="Place image"
            className="place-card__image"
            height={200}
            src="img/1.png"
            width={260}
          />
        </a>
      </div>
      <div
        className="place-card__info"
      >
        <div
          className="place-card__price-wrapper"
        >
          <div
            className="place-card__price"
          >
            <b
              className="place-card__price-value"
            >
              €
              120
            </b>
            <span
              className="place-card__price-text"
            >
              / night
            </span>
          </div>
          <button
            className="cities__bookmark-button button "
            onClick={[Function]}
            type="button"
          >
            <svg
              className="place-card__bookmark-icon"
              height={19}
              width={18}
            >
              <use
                xlinkHref="#icon-bookmark"
              />
            </svg>
            <span
              className="visually-hidden"
            >
              To bookmarks
            </span>
          </button>
        </div>
        <div
          className="place-card__rating rating"
        >
          <div
            className="place-card__stars rating__stars"
          >
            <span
              style={
                Object {
                  "width": "100%",
                }
              }
            />
            <span
              className="visually-hidden"
            >
              Rating
            </span>
          </div>
        </div>
        <h2
          className="place-card__name"
        >
          <a
            href="/offer/4"
            onClick={[Function]}
          >
            Beautiful & luxurious studio at great location
          </a>
        </h2>
        <p
          className="place-card__type"
        >
          apartment
        </p>
      </div>
    </article>
  </div>
</section>
`;

exports[`Render places (modificator - near) correctly 1`] = `
<section
  className="places near-places"
>
  <h2
    className="near-places__title"
  >
    Other places in the neighbourhood
  </h2>
  <div
    className="places__list near-places__list"
  >
    <article
      className="cities__place-card place-card"
      onMouseEnter={[Function]}
      onMouseLeave={[Function]}
    >
      <div
        className="cities__image-wrapper place-card__image-wrapper"
      >
        <a
          href="#"
        >
          <img
            alt="Place image"
            className="place-card__image"
            height={200}
            src="img/1.png"
            width={260}
          />
        </a>
      </div>
      <div
        className="place-card__info"
      >
        <div
          className="place-card__price-wrapper"
        >
          <div
            className="place-card__price"
          >
            <b
              className="place-card__price-value"
            >
              €
              120
            </b>
            <span
              className="place-card__price-text"
            >
              / night
            </span>
          </div>
          <button
            className="cities__bookmark-button button "
            onClick={[Function]}
            type="button"
          >
            <svg
              className="place-card__bookmark-icon"
              height={19}
              width={18}
            >
              <use
                xlinkHref="#icon-bookmark"
              />
            </svg>
            <span
              className="visually-hidden"
            >
              To bookmarks
            </span>
          </button>
        </div>
        <div
          className="place-card__rating rating"
        >
          <div
            className="place-card__stars rating__stars"
          >
            <span
              style={
                Object {
                  "width": "100%",
                }
              }
            />
            <span
              className="visually-hidden"
            >
              Rating
            </span>
          </div>
        </div>
        <h2
          className="place-card__name"
        >
          <a
            href="/offer/1"
            onClick={[Function]}
          >
            Beautiful & luxurious studio at great location
          </a>
        </h2>
        <p
          className="place-card__type"
        >
          apartment
        </p>
      </div>
    </article>
    <article
      className="cities__place-card place-card"
      onMouseEnter={[Function]}
      onMouseLeave={[Function]}
    >
      <div
        className="cities__image-wrapper place-card__image-wrapper"
      >
        <a
          href="#"
        >
          <img
            alt="Place image"
            className="place-card__image"
            height={200}
            src="img/1.png"
            width={260}
          />
        </a>
      </div>
      <div
        className="place-card__info"
      >
        <div
          className="place-card__price-wrapper"
        >
          <div
            className="place-card__price"
          >
            <b
              className="place-card__price-value"
            >
              €
              120
            </b>
            <span
              className="place-card__price-text"
            >
              / night
            </span>
          </div>
          <button
            className="cities__bookmark-button button "
            onClick={[Function]}
            type="button"
          >
            <svg
              className="place-card__bookmark-icon"
              height={19}
              width={18}
            >
              <use
                xlinkHref="#icon-bookmark"
              />
            </svg>
            <span
              className="visually-hidden"
            >
              To bookmarks
            </span>
          </button>
        </div>
        <div
          className="place-card__rating rating"
        >
          <div
            className="place-card__stars rating__stars"
          >
            <span
              style={
                Object {
                  "width": "100%",
                }
              }
            />
            <span
              className="visually-hidden"
            >
              Rating
            </span>
          </div>
        </div>
        <h2
          className="place-card__name"
        >
          <a
            href="/offer/2"
            onClick={[Function]}
          >
            Beautiful & luxurious studio at great location
          </a>
        </h2>
        <p
          className="place-card__type"
        >
          apartment
        </p>
      </div>
    </article>
    <article
      className="cities__place-card place-card"
      onMouseEnter={[Function]}
      onMouseLeave={[Function]}
    >
      <div
        className="cities__image-wrapper place-card__image-wrapper"
      >
        <a
          href="#"
        >
          <img
            alt="Place image"
            className="place-card__image"
            height={200}
            src="img/1.png"
            width={260}
          />
        </a>
      </div>
      <div
        className="place-card__info"
      >
        <div
          className="place-card__price-wrapper"
        >
          <div
            className="place-card__price"
          >
            <b
              className="place-card__price-value"
            >
              €
              120
            </b>
            <span
              className="place-card__price-text"
            >
              / night
            </span>
          </div>
          <button
            className="cities__bookmark-button button "
            onClick={[Function]}
            type="button"
          >
            <svg
              className="place-card__bookmark-icon"
              height={19}
              width={18}
            >
              <use
                xlinkHref="#icon-bookmark"
              />
            </svg>
            <span
              className="visually-hidden"
            >
              To bookmarks
            </span>
          </button>
        </div>
        <div
          className="place-card__rating rating"
        >
          <div
            className="place-card__stars rating__stars"
          >
            <span
              style={
                Object {
                  "width": "100%",
                }
              }
            />
            <span
              className="visually-hidden"
            >
              Rating
            </span>
          </div>
        </div>
        <h2
          className="place-card__name"
        >
          <a
            href="/offer/3"
            onClick={[Function]}
          >
            Beautiful & luxurious studio at great location
          </a>
        </h2>
        <p
          className="place-card__type"
        >
          apartment
        </p>
      </div>
    </article>
    <article
      className="cities__place-card place-card"
      onMouseEnter={[Function]}
      onMouseLeave={[Function]}
    >
      <div
        className="cities__image-wrapper place-card__image-wrapper"
      >
        <a
          href="#"
        >
          <img
            alt="Place image"
            className="place-card__image"
            height={200}
            src="img/1.png"
            width={260}
          />
        </a>
      </div>
      <div
        className="place-card__info"
      >
        <div
          className="place-card__price-wrapper"
        >
          <div
            className="place-card__price"
          >
            <b
              className="place-card__price-value"
            >
              €
              120
            </b>
            <span
              className="place-card__price-text"
            >
              / night
            </span>
          </div>
          <button
            className="cities__bookmark-button button "
            onClick={[Function]}
            type="button"
          >
            <svg
              className="place-card__bookmark-icon"
              height={19}
              width={18}
            >
              <use
                xlinkHref="#icon-bookmark"
              />
            </svg>
            <span
              className="visually-hidden"
            >
              To bookmarks
            </span>
          </button>
        </div>
        <div
          className="place-card__rating rating"
        >
          <div
            className="place-card__stars rating__stars"
          >
            <span
              style={
                Object {
                  "width": "100%",
                }
              }
            />
            <span
              className="visually-hidden"
            >
              Rating
            </span>
          </div>
        </div>
        <h2
          className="place-card__name"
        >
          <a
            href="/offer/4"
            onClick={[Function]}
          >
            Beautiful & luxurious studio at great location
          </a>
        </h2>
        <p
          className="place-card__type"
        >
          apartment
        </p>
      </div>
    </article>
  </div>
</section>
`;
